src/controllers/AddAssetsManager.js

```diff
@@ -165,25 +179,55 @@ export default class WatchAssetManager extends EventEmitter {
   }
 
   async normalizeWatchAssetParams(assetId, assetParams, providerConfig, web3) {
-    const { address, decimals, symbol } = assetParams.options || {}
-    if (!address) throw ethErrors.rpc.invalidParams('Invalid watch asset params: asset address is required.')
-    const tokenHandler = new TokenHandler({ address: address.toLowerCase(), web3 })
-    const options = assetParams.options || {}
-    if (!decimals) options.decimals = await tokenHandler.getDecimals()
-    if (!symbol) options.symbol = await tokenHandler.getSymbol()
-    const name = await tokenHandler.getName()
-    const metadata = {
-      network: providerConfig.host,
-      name,
+    const nft_contract_standard = [CONTRACT_TYPE_ERC721, CONTRACT_TYPE_ERC1155]
+    let finalParams = {}
+    if (assetParams.type.toLowerCase() === CONTRACT_TYPE_ERC20) {
+      const { address, decimals, symbol } = assetParams.options || {}
+      if (!address) throw ethErrors.rpc.invalidParams('Invalid watch asset params: asset address is required.')
+      const tokenHandler = new TokenHandler({ address: address.toLowerCase(), web3 })
+      const options = assetParams.options || {}
+      if (!decimals) options.decimals = await tokenHandler.getDecimals()
+      if (!symbol) options.symbol = await tokenHandler.getSymbol()
+      const name = await tokenHandler.getName()
+      const metadata = {
+        network: providerConfig.host,
+        name,
+      }
+      finalParams = {
+        id: assetId,
+        ...assetParams,
+        options: {
+          ...assetParams.options,
+          ...options,
+        },
+        metadata,
+      }
+      // return finalParams
     }
-    const finalParams = {
-      id: assetId,
-      ...assetParams,
-      options: {
-        ...assetParams.options,
-        ...options,
-      },
-      metadata,
+    if (nft_contract_standard.includes(assetParams.type.toLowerCase())) {
+      const { address, id, image, name, balance } = assetParams.options || {}
+      const userAddress = this.prefsController.store.getState().selectedAddress
+      log.info(userAddress)
+      if (!address) throw ethErrors.rpc.invalidParams('Invalid watch asset params: asset address is required.')
```

Comment:
error message is incorrect, its user address, not asset address.

Reply:
it's correct, user address is fetched from vuex store, address is taken input from user.

Reply:
right, its fine. just move the error msg above userAddress.